roles/spark/templates/spark-defaults.conf.j2

```diff
@@ -7,7 +7,8 @@
 # spark.eventLog.dir               hdfs://namenode:8021/directory
 # spark.serializer                 org.apache.spark.serializer.KryoSerializer
 # spark.driver.memory              5g
-# spark.executor.extraJavaOptions  -XX:+PrintGCDetails -Dkey=value -Dnumbers="one two three"
+spark.executor.extraJavaOptions -XX:{{ gc_type }} -XX:+UseCompressedOops {{ gc_debug }} -agentlib:jdwp=transport=dt_socket,server=y,suspend={{ worker_waiting_on_startup }},address={{ worker_debug_port }}
```

Comment:
Does this mean that the spark drivers and executors will always run in debug mode? 
If so there could be a performance hit with running in debug mode. 
Maybe comment out the `-agentlib...` and to spark.md#debug mode instruct to uncomment.

Reply:
I have added spark_debug_mode global variable. If you set to true the debug information will be printed and Spark will run in debug mode.